metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/query/ESBrowseDAO.java

```diff
@@ -74,14 +76,26 @@ public BrowseResult browse(@Nonnull String entityName, @Nonnull String path, @Nu
 
     try {
       final String indexName = indexConvention.getIndexName(entityRegistry.getEntitySpec(entityName));
+
       final SearchResponse groupsResponse =
-          client.search(constructGroupsSearchRequest(indexName, path, requestMap, 0, 1000), RequestOptions.DEFAULT);
+          client.search(constructGroupsSearchRequest(indexName, path, requestMap), RequestOptions.DEFAULT);
+      final BrowseResultMetadata browseResultMetadata = extractGroupsResponse(groupsResponse, path, from, size);
+      final int numGroups = browseResultMetadata.getTotalNumEntities().intValue();
+
+      // Based on the number of groups returned, compute the from and size to query for entities
+      int entityFrom = Math.max(from - numGroups, 0);
+      int entitySize = Math.min(Math.max(from + size - numGroups, 0), size);
       final SearchResponse entitiesResponse =
-          client.search(constructEntitiesSearchRequest(indexName, path, requestMap, from, size),
+          client.search(constructEntitiesSearchRequest(indexName, path, requestMap, entityFrom, entitySize),
               RequestOptions.DEFAULT);
```

Comment:
small optimization available here- if entitySize is 0 we don't have to do this query

Reply:
or do we need to for counts?

Reply:
Yeah. I was going to do that until you mentioned we need the whole counts